index.html

```diff
@@ -722,56 +724,61 @@ <h2>
             <ol>
-              <li>
-                <span id="wait-for-change">Wait for a significant change of
-                geographic position</span>. What constitutes a significant
-                change of geographic position is left to the implementation.
-                User agents MAY impose a rate limit on how frequently position
-                changes are reported.
+              <li>Set |permission| to [=request permission to use=]
+              |descriptor|.
               </li>
-              <li>If |document| is not [=Document/fully active=] or
-              [=Document/visibility state=] is not "visible", go back to the
-              previous step and again <a href="#wait-for-change">wait for a
-              significant change of geographic position</a>.
-                <aside class="note" title=
-                "Position updates are exclusively for fully-active visible documents">
-                  <p>
-                    The desired effect here being that position updates are
-                    exclusively delivered to fully active documents that are
-                    visible; Otherwise the updates get silently "dropped on the
-                    floor". Only once a document again becomes fully active and
-                    visible (e.g., an [^iframe^] gets reattached to a parent
-                    document), do the position updates once again start getting
-                    delivered.
-                  </p>
-                </aside>
+              <li data-tests=
+              "getCurrentPosition_permission_allow.https.html, getCurrentPosition_permission_deny.https.html">
+              If |permission| is "denied", then:
+                <ol>
+                  <li>If |watchId| was passed, [=list/remove=] |watchId| from
+                  |watchIDs|.
+                  </li>
+                  <li>[=Call back with error=] passing |errorCallback| and
+                  {{GeolocationPositionError/PERMISSION_DENIED}}.
+                  </li>
+                  <li>Terminate this algorithm.
+                  </li>
+                </ol>
               </li>
               <li>Wait to [=acquire a position=] passing |successCallback|,
               |errorCallback|, |options|, and |watchId|.
               </li>
+              <li>If |watchId| was not passed, terminate this algorithm.
+              </li>
+              <li>While |watchIDs| [=list/contains=] |watchId|:
+                <ol>
+                  <li>
+                    <span id="wait-for-change">Wait for a significant change of
+                    geographic position</span>. What constitutes a significant
+                    change of geographic position is left to the
+                    implementation. User agents MAY impose a rate limit on how
+                    frequently position changes are reported.
+                  </li>
+                  <li>If |document| is not [=Document/fully active=] or
```

Comment:
Can we check this while "in parallel"? If not then we should do a larger rewrite (perhaps in a follow-up to reduce churn) to only go "in parallel" for specifically the requesting permission and acquiring a position steps.

Reply:
yeah, good point. We should probably just queue a global task here that can then check the state of the document sync.